### PR TITLE
feat(reader): remove ParseNode infrastructure

### PR DESCRIPTION
**Pull request by baywet**
## Summary
- remove the internal ParseNode hierarchy in favor of direct System.Text.Json.Nodes.JsonNode parsing
- move map/list/value parsing behavior into JsonNodeHelper with explicit ParsingContext propagation
- update V1/V1_1 deserializers and tests to use JsonNode directly

## Validation
- dotnet test
- dotnet format

### CODE REVIEW

File: src/lib/Reader/JsonNodeHelper.cs [hunk 1]
```diff
@@ -1,22 +1,167 @@
...
+                value
+            };
+        });
+
+        return nodes.ToDictionary(k => k.key, v => v.value);
+    }
+
+    public static JsonNode CreateAny(this JsonNode? node)
+    {
+        return node ?? JsonValue.Create((string?)null)!;
```
Comment (baywet) [inline src/lib/Reader/JsonNodeHelper.cs]: This behaviour is odd and adds a potential to introduce errors. We should add a test case that the library is actually able to apply "null" to something. Otherwise, refactor to avoid using a bang operator.

**Review by kilifu — APPROVED** [review]
